README.md

```diff
@@ -0,0 +1,78 @@
+# SSV Distributed Key Generation Spec
+
+## Introduction
+This repository contains spec of distributed key generation (DKG) used in SSV. Based on verifiable secret sharing, DKG allows a set of *n* operators to generate a  **BLS** key share for each operator, such that a BLS private key can only be used when threshold $t$ out of $n$ operators agree. $t$ is defined by the number of faults admissible: $t = n - f$, where $n = 3 * f + 1$.
+
+---
+
+## Functionalities
+Operations are initiated by an initiator, and executed by operators. An initiator can initiate:
+
+### Init
+When a init message is sent to the operators, all operators participate and run a DKG ceremony to create key shares. A DKG ceremony asks each operator to verifiably share locally generated secrets with other operators, and combine secret shares from all operators to derive its own BLS key share. After the ceremony, each operator generates a **Result** with a **SignedProof** (see below) for verification. Proofs are referred to when an initiator asks for re-sign or reshare. 
+
+### Re-sign
+In the case where the nonces of a DKG ceremony is not correct, to prevent replay attack, the initiator sends re-sign messages for operators to create new signatures with correct nonces without generating a new validator key. 
+
+### Reshare
+When the set of operators changed (including change of size and change of operators), a reshare is initiated to generate new key shares among the new operators. This process requires participation from *t* old operators and all new operators. 
+
+---
```

Comment:
Add the following here (or wherever you see fit)

## Security Assumptions

All messages are authenticated using the initiator's and operators' public key.

### Init
The initiator is fully trusted.

### Re-sign
The initiator is not trusted. It may try to fool the operators into resigning with the wrong secret shares. Also since in the resign operation the `owner` may change, and `owner` is used in the `Proof`, all $n$ operators need to participate in the ceremony.

### Reshare
The initiator is not trusted. It may try to fool the operators into resharing with the wrong secret shares.

## Validations
In a DKG ceremony, each message sent between the initiator and the operators are validated by the operators such that:
- The message is signed by the sender, preventing spoofing attacks.
- Where replay attacks are possible, a valid nonce is included to ensure the message is fresh (otherwise **Re-sign**).
- The message is in the same scope, i.e., the message is regarding the same ceremony, the same set of operators and it is in the correct stage as the receiver expects.